ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/CustomParser_Test.java

```diff
@@ -0,0 +1,172 @@
+                .getBulkJobSpectraS3(new GetBulkJobSpectraS3Request(BUCKET_NAME, Lists.newArrayList(object)));
+
+        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        final GetObjectRequest request = new GetObjectRequest(
+                BUCKET_NAME,
+                OBJ_NAME,
+                getBulkJobSpectraS3Response.getMasterObjectList().getJobId(),
+                0,
+                stream);
+
+        client.getObject(request, new Function<GetObjectCustomParserParameters, GetObjectResponse>() {
+            @Override
+            public GetObjectResponse apply(GetObjectCustomParserParameters getObjectParserParams) {
+                final WebResponse webResponse = getObjectParserParams.getWebResponse();
+                if (ResponseParserUtils.validateStatusCode(webResponse.getStatusCode(), 200, 206)) {
+                    final  long objectSize = getSizeFromHeaders(webResponse.getHeaders());
+                    try (final InputStream responseStream = webResponse.getResponseStream()) {
+                        final long totalBytes = IOUtils.copy(
+                                responseStream,
+                                getObjectParserParams.getDestinationChannel(),
+                                getObjectParserParams.getBufferSize(),
+                                getObjectParserParams.getObjectName(),
+                                false);
+                        getObjectParserParams.getDestinationChannel().close();
+
+                        if (objectSize != -1 && totalBytes != objectSize) {
+                            throw new ContentLengthNotMatchException(getObjectParserParams.getObjectName(), objectSize, totalBytes);
+                        }
+
+                    } catch (final IOException e) {
+                        fail();
+                    }
+                }
+                return null;
+            }
+        });
+        assertThat(stream.toByteArray(), is(OBJ_BYTES));
+    }
+
+    @SuppressWarnings("deprecation")
+    @Test
+    public void getObjectCustomParserNoObject_Test() throws IOException{
+        final String objectName = "doesNotExistObject";
+
+        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        final ObjectOutputStreamBuilder builder = new ObjectOutputStreamBuilder() {
+            @Override
+            public OutputStream buildOutputStream(String key) throws IOException {
+                return stream;
+            }
+        };
+        final GetObjectRequest request = new GetObjectRequest(
+                BUCKET_NAME,
+                objectName,
+                builder.buildChannel(objectName));
+
+        //Get object that does not exist
+        try {
+            client.getObject(request, new Function<GetObjectCustomParserParameters, GetObjectResponse>() {
+                @Override
+                public GetObjectResponse apply(GetObjectCustomParserParameters getObjectParserParams) {
```

Comment:
My assumption is that the call would have failed before getting to this point.  If that's not the case, the GetObjectCustomParser needs to handle this.